PR review

mkrout opened the pull request.

mkrout committed:
-Add lead management portlet

mkrout committed:
-Add lead management Model / services

mkrout committed:
-Add lead management Model / services

mkrout committed:
-Add lead management Model / services

mkrout committed:
-Review Vue application

mkrout committed:
-Add mail templates management

mkrout committed:
-Add mail templates management

mkrout committed:
-Add unit test first iteration

mkrout committed:
-Add unit test second iteration

mkrout committed:
-Add Edit Lead

mkrout committed:
-Add mail templates management

mkrout committed:
-Synchronize with Task

mkrout committed:
-Synchronize with Task

mkrout committed:
-Add Lead Capture Settings

mkrout committed:
-Add Lead Capture Settings

mkrout committed:
-Fix exception when creating user

mkrout committed:
-Add logs

mkrout committed:
- review

mkrout committed:
-Add security token to add leads rest service

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/dao/MailTemplateDAO.java:
```diff
@@ -0,0 +1,34 @@
+    } catch (NoResultException e) {
+      return null;
+    } catch (Exception e) {
+      LOG.error("Error occurred when trying to get list of mail templates by evant {}", event, e);
```
```suggestion
      LOG.error("Error occurred when trying to get list of mail templates by event {}", event, e);
```

mkrout committed:
-Add security token to add leads rest service

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/dao/FieldDAO.java:
```diff
@@ -0,0 +1,31 @@
+      return null;
+    } catch (Exception e) {
+      LOG.error("Error occurred when trying to get list of fields by formId {}", responseId, e);
+      return null;
```
Is a null value will be handled everywhere the method is called ?
(the question is applicable to all the dao's methods)

mkrout replied:
yep

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/dao/FieldDAO.java:
```diff
@@ -0,0 +1,31 @@
+    try {
+      return query.getResultList();
+    } catch (NoResultException e) {
+      return null;
```
Is a null value will be handled everywhere the method is called ?
(the question is applicable to all the dao's methods)

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/entity/FieldEntity.java:
```diff
@@ -0,0 +1,45 @@
+@Table(name = "ADDONS_LC_FORM_FIELD")
+@Data
+@NamedQueries({
+    @NamedQuery(name = "FieldEntity.getFileldsByResponse", query = "SELECT field FROM FieldEntity field where field.responseEntity.id = :responseId ") })
```
```suggestion
    @NamedQuery(name = "FieldEntity.getFieldsByResponse", query = "SELECT field FROM FieldEntity field where field.responseEntity.id = :responseId ") })
```

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/listeners/NewResponseListener.java:
```diff
@@ -0,0 +1,71 @@
+        content = Utils.getContentForMail(mailTemplateDTO, lead);
+        if (content != null && shouldBeSend(template, responseEntity)) {
+          lcMailService.sendMail(content.getContent(), content.getSubject(), lead);
+          LOG.info("Mail sent to {}",lead.getMail());
```
* The id of the user or the lead should be logged instead of the email to respect the RGPD containts
* the reason NewResponse and the templateId could also be useful on the log message

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/listeners/NewLeadListener.java:
```diff
@@ -0,0 +1,111 @@
+      if (users.getSize() > 0) {
+        User communityUser = users.load(0, 1)[0];
+        lead.setCommunityUserName(communityUser.getUserName());
+        lead.setCommunityRegistrationDate(communityUser.getCreatedDate().getTime());
```
A log to indicate the association between the lead and the user should be added.

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/listeners/NewLeadListener.java:
```diff
@@ -0,0 +1,111 @@
+    lead.setTaskId(task.getId());
+    lead.setTaskUrl(TaskUtil.buildTaskURL(task));
+    leadDAO.update(lead);
+
```
A functional info log with the task id should be added

mkrout replied:
Ok

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/listeners/NewLeadListener.java:
```diff
@@ -0,0 +1,111 @@
+      if (mailTemplateDTO.getContents().size() > 0) {
+        content = Utils.getContentForMail(mailTemplateDTO, lead);
+        if (content != null) {
+          lcMailService.sendMail(content.getContent(), content.getSubject(), lead);
```
A new log with the templateid leadid and reason (NewLead) could also be useful

mkrout replied:
Ok

vsellier replied:
👍

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/listeners/NewUserListener.java:
```diff
@@ -0,0 +1,62 @@
+          leadEntity.setCommunityRegistration(true);
+          leadEntity.setCommunityRegistrationDate(user.getCreatedDate().getTime());
+          leadEntity.setCommunityUserName(user.getUserName());
+          leadDAO.update(leadEntity);
```
A new log with detail of the association could be useful

mkrout replied:
Ok

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/listeners/NewUserListener.java:
```diff
@@ -0,0 +1,62 @@
+        }
+      }
+    } catch (Exception e) {
+      LOG.error("an error accured", e);
```
```suggestion
      LOG.error("an error occured", e);
```

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/listeners/NewUserListener.java:
```diff
@@ -0,0 +1,62 @@
+        }
+      }
+    } catch (Exception e) {
+      LOG.error("an error accured", e);
```
You should add the userId if available at this step

vsellier replied:
the remark is still true

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/listeners/TaskUpdateListener.java:
```diff
@@ -0,0 +1,51 @@
+             }
+             if (isDiff(before.getStatus(), after.getStatus())) {
+                 leadEntity.setStatus(after.getStatus().getName());
+             }
```
A new log with the detail of the action could be useful

mkrout replied:
Ok

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadCaptureSettingsRest.java:
```diff
@@ -0,0 +1,118 @@
+    try {
+      return Response.ok(leadCaptureSettingsService.getSettings(), mediaType).build();
+    } catch (Exception e) {
+      return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
```
An error should be logged

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadCaptureSettingsRest.java:
```diff
@@ -0,0 +1,118 @@
+    MediaType mediaType = RestChecker.checkSupportedFormat("json", SUPPORTED_FORMATS);
+    try {
+      leadCaptureSettingsService.saveSettings(settings);
+      LOG.info("Lead capture settings updated by {}",sourceIdentity.getRemoteId());      return Response.ok("settingsUpdated", mediaType).build();
```
code format

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadCaptureSettingsRest.java:
```diff
@@ -0,0 +1,118 @@
+  @POST
+  @Path("settings")
+  public Response save(@Context UriInfo uriInfo, LeadCaptureSettings settings) throws Exception {
+    Identity sourceIdentity = Util.getAuthenticatedUserIdentity(portalContainerName);
```
This action should be allowed only to an adminitrator

mkrout replied:
yep

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadCaptureSettingsRest.java:
```diff
@@ -0,0 +1,118 @@
+      leadCaptureSettingsService.saveSettings(settings);
+      LOG.info("Lead capture settings updated by {}",sourceIdentity.getRemoteId());      return Response.ok("settingsUpdated", mediaType).build();
+    } catch (Exception e) {
+      return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
```
An error should be logged

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadCaptureSettingsRest.java:
```diff
@@ -0,0 +1,118 @@
+    }
+    MediaType mediaType = RestChecker.checkSupportedFormat("json", SUPPORTED_FORMATS);
+    try {
+      return Response.ok(leadCaptureSettingsService.getSettings(), mediaType).build();
```
Is it really necessary to expose all the settings to all the users ?

mkrout replied:
No, it should be exposed only for administrators

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadCaptureSettingsRest.java:
```diff
@@ -0,0 +1,118 @@
+      context.put("leadCaptureConfigured", isConfigured(settings));
+      return Response.ok(context.toString(), mediaType).build();
+    } catch (Exception e) {
+      return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
```
An error should be logged

mkrout replied:
Ok

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java:
```diff
@@ -0,0 +1,227 @@
+  @Path("leads")
+  public Response getLeads(@Context UriInfo uriInfo) throws Exception {
+    Identity sourceIdentity = Util.getAuthenticatedUserIdentity(portalContainerName);
+    if (sourceIdentity == null) {
```
It should not be exposed to all the users

mkrout replied:
Now it's exposed only to ux-team group members

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java:
```diff
@@ -0,0 +1,227 @@
+    try {
+      return Response.ok(leadsManagementService.getLeads(), mediaType).build();
+    } catch (Exception e) {
+      return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
```
An error should be logged

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java:
```diff
@@ -0,0 +1,227 @@
+  public Response add(@Context UriInfo uriInfo,@HeaderParam("token") String headerToken, FormInfo lead) throws Exception {
+    String captureToken = leadCaptureSettingsService.getSettings().getLeadCaptureToken();
+    if (headerToken == null || captureToken==null || !captureToken.equals(headerToken)) {
+      LOG.warn("Access frobidden to the add lead rest service, wrong token: {}",headerToken);
```
```suggestion
      LOG.warn("Access forbidden to the add lead rest service, wrong token: {}", headerToken);
```

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java:
```diff
@@ -0,0 +1,227 @@
+      return Response.status(Response.Status.FORBIDDEN).build();
+    }
+    if (!leadCaptureSettingsService.getSettings().isCaptureEnabled()) {
+      LOG.warn("Lead capture not enabled, New lead {} not captured ",lead.getLead().getMail());
```
The mail should not be logged directly on the log to match the RGPD contraints

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java:
```diff
@@ -0,0 +1,227 @@
+      MediaType mediaType = RestChecker.checkSupportedFormat("json", SUPPORTED_FORMATS);
+      try {
+        leadsManagementService.addLeadInfo(lead, true);
+        LOG.info("service=lead-capture operation=synchronize_lead parameters=\"lead_mail:{},form_name:{}\"", lead.getLead().getMail(), lead.getResponse().getFormName());
```
The email or personnal data should not be logged directly for RGPD contraints

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java:
```diff
@@ -0,0 +1,227 @@
+        LOG.info("service=lead-capture operation=synchronize_lead parameters=\"lead_mail:{},form_name:{}\"", lead.getLead().getMail(), lead.getResponse().getFormName());
+        return Response.ok("lead synchronized", mediaType).build();
+      } catch (Exception e) {
+        return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
```
An error should be logged, It's valid for any error reply

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java:
```diff
@@ -0,0 +1,227 @@
+  @Path("leads/{id}")
+  public Response deletelead(@Context UriInfo uriInfo, @PathParam("id") Long id) throws Exception {
+    Identity sourceIdentity = Util.getAuthenticatedUserIdentity(portalContainerName);
+    if (sourceIdentity == null) {
```
It should not be accessible to any connected user, the same remark is valid for all the entry points

mkrout replied:
Now it's exposed only to ux-team group members

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java:
```diff
@@ -0,0 +1,227 @@
+
+  @DELETE
+  @Path("leads/{id}")
+  public Response deletelead(@Context UriInfo uriInfo, @PathParam("id") Long id) throws Exception {
```
```suggestion
  public Response deleteLead(@Context UriInfo uriInfo, @PathParam("id") Long id) throws Exception {
```

ahamdi commented on pom.xml:
```diff
@@ -0,0 +1,104 @@
+        <url>https://github.com/exo-addons/lead-capture</url>
+    </scm>
+    <properties>
+        <platform.version>6.0.0-M09</platform.version>
```
```suggestion
        <platform.version>6.0.x-SNAPSHOT</platform.version>
```

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java:
```diff
@@ -0,0 +1,227 @@
+  }
+
+  @PUT
+  @Path("leads/{id}")
```
Isn't the common rule to use the singular to hit an identified uniq resource ?

mkrout replied:
Don't think so, according to REST guidelines, REST endpoints nouns must be always plural, .

ahamdi commented on services/src/main/java/org/exoplatform/leadcapture/Utils.java:
```diff
@@ -0,0 +1,211 @@
+  public static final String EMPTY_STR                                     = "";
+  public static final SimpleDateFormat formatter                           = new SimpleDateFormat(DATE_FORMAT);
+  public static final String LEAD_CAPTURE_SCOPE_NAME                       = "ADDONS_LEAD_CAPTURE";
+  public static final String LEAD_CAPTURE_CONTEXT_NAME                     = "ADDONS_LEAD_CAPTURE";
```
Is there a real need for declaring two constants for the same String ?

ahamdi commented on services/src/main/java/org/exoplatform/leadcapture/Utils.java:
```diff
@@ -0,0 +1,211 @@
+        activity.setUserId(posterIdentity.getId());
+        return activityManager.saveActivity(spaceIdentity, activity);
+      } else {
+        LOG.warn("Not able to create the activity, the Poster or Space Identity is missing");
```
Could you specify which variable is Null in the log to simplify problem detection

mkrout replied:
Yep

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java:
```diff
@@ -0,0 +1,227 @@
+        return Response.status(Response.Status.NOT_FOUND).entity("Lead Not found").build();
+      }
+      leadsManagementService.deleteLead(lead);
+      LOG.info("Lead {} deleted by {}", lead.getMail(), sourceIdentity.getRemoteId());
```
The email should not be logged for rgpd constraints, please replace everywhere it's logged

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java:
```diff
@@ -0,0 +1,227 @@
+  }
+
+  @POST
+  @Path("leads")
```
As this method is for a unique lead, wouldn't the singular be better?

mkrout replied:
Don't think so, according to REST guidelines, REST endpoints nouns must be always plural, .

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/MailTemplatesManagementRest.java:
```diff
@@ -0,0 +1,95 @@
+import org.exoplatform.social.service.rest.RestChecker;
+import org.exoplatform.social.service.rest.Util;
+
+@Path("/leadcapture/mailtemplatesmanagement")
```
Same remark for securisation

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/Utils.java:
```diff
@@ -0,0 +1,211 @@
+  public static final JsonParser JSON_PARSER                               = new JsonParserImpl();
+  public static final JsonGenerator JSON_GENERATOR                         = new JsonGeneratorImpl();
+
+  public static JSONObject toResponseJson(ResponseEntity responseEntity) throws JSONException {
```
The logs of this class should be reviewed to add pertinent information, mostly the id of the objects causing the error

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/listeners/NewLeadListener.java:
```diff
@@ -0,0 +1,111 @@
+      }
+    }
+
+    ExoSocialActivity activity = Utils.createActivity(lead);
```
Be careful to the NPE

mkrout replied:
Good catch thanx

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/Utils.java:
```diff
@@ -0,0 +1,211 @@
+      commentJson.put(CREATION_DATE_FIELD_NAME, Utils.getFormatter().format(comment.getCreatedTime()));
+      return commentJson;
+    } catch (Exception e) {
+      LOG.error("Cannot conevert comment to json", e);
```
```suggestion
      LOG.error("Cannot convert comment to json", e);
```

vsellier commented on services/src/main/resources/db/changelog/lead-capture.db.changelog-1.0.0.xml:
```diff
@@ -0,0 +1,180 @@
+            <column name="ID" type="BIGINT" autoIncrement="${autoIncrement}" startWith="1">
+                <constraints nullable="false" primaryKey="true" primaryKeyName="PK_LC_LEAD"/>
+            </column>
+            <column name="MAIL" type="NVARCHAR(150)">
```
should be ate least 256 : https://tools.ietf.org/html/rfc5321#section-4.5.3

mkrout replied:
Ok

vsellier commented on services/src/main/resources/db/changelog/lead-capture.db.changelog-1.0.0.xml:
```diff
@@ -0,0 +1,180 @@
+            </column>
+            <column name="BLOG_SUBSCRIPTION_DATE" type="BIGINT">
+            </column>
+            <column name="COMMUNITY_USER_NAME" type="NVARCHAR(150)">
```
If it's the remote_id, the size should be 200

vsellier commented on services/src/main/resources/db/changelog/lead-capture.db.changelog-1.0.0.xml:
```diff
@@ -0,0 +1,180 @@
+
+    </changeSet>
+
+    <changeSet author="exo-lead-capture" id="1.0.0-2" dbms="oracle,postgresql">
```
Is the Oracle compatibility really necessary ?

mkrout replied:
No, we need only postgres compatibility .

vsellier commented on pom.xml:
```diff
@@ -0,0 +1,104 @@
+        <platform.version>6.0.0-M09</platform.version>
+        <org.lombok.version>1.18.2</org.lombok.version>
+        <org.jackson.version>2.4.2</org.jackson.version>
+        <addon.exo.tasks.version>3.0.x-SNAPSHOT</addon.exo.tasks.version>
```
It should be coherent with the plaform.version property

mkrout replied:
Ok

vsellier commented on packaging/pom.xml:
```diff
@@ -0,0 +1,53 @@
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>lead-capture-war</artifactId>
+            <version>1.0.x-SNAPSHOT</version>
```
Is this version really needed ?

mkrout replied:
No, good catch

vsellier commented:
@mkrout Have you checked if the email will only be sent to the usual restricted recipient list on the environment other than the production

vsellier reviewed (changes requested):
@mkrout Have you checked if the email will only be sent to the usual restricted recipient list on the environment other than the production

ahamdi commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java:
```diff
@@ -0,0 +1,227 @@
+    if (sourceIdentity == null) {
+      return Response.status(Response.Status.UNAUTHORIZED).build();
+    }
+    MediaType mediaType = RestChecker.checkSupportedFormat("json", SUPPORTED_FORMATS);
```
COuld be replaced with javax.ws.rs.core.MediaType#APPLICATION_JSON_TYPE

ahamdi replied:
OK

mkrout committed:
Apply Vincent's suggestions

Co-Authored-By: Vincent Sellier <vincent.sellier@gmail.com>

mkrout committed:
- Code review

mkrout committed:
- Code review

mkrout committed:
- Code review
- Use Timestamp instead of BIGINT for dates
- Add ux-team group
- secure rest api

mkrout committed:
-

mkrout committed:
- fix settings

ahamdi commented on services/src/main/java/org/exoplatform/leadcapture/services/LeadsManagementService.java:
```diff
@@ -0,0 +1,481 @@
+        fieldDAO.deleteAll(fieldDAO.getFieldsByResponse(responseEntity.getId()));
+      }
+      responseDAO.deleteAll(responseEntities);
+      leadDAO.delete(lead);
```
Are you sure that deleting a lead is needed ? could we make it as hidden somehow (just adding a deleted flag)?

mkrout replied:
Yes, no need to hide leads,  we cna delete them when needed

ahamdi commented on war/src/main/webapp/WEB-INF/web.xml:
```diff
@@ -0,0 +1,61 @@
+
+    <filter-mapping>
+        <filter-name>ThreadLocalSessionProviderInitializedFilter</filter-name>
+        <url-pattern>/*</url-pattern>
```
I believe this is not needed for the extension

mkrout replied:
right

ahamdi replied:
All filters should be removed except ResourceRequestFilter

mkrout replied:
Okey

mkrout committed:
- fix settings
- Add cross origin conf

mkrout committed:
- fix settings
- Add cross origin conf

mkrout committed:
- Code review

mkrout committed:
- Code review

mkrout committed:
- Fix create lead without responses

mkrout commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java:
```diff
@@ -0,0 +1,281 @@
+    if (captureToken == null || !captureToken.equals(headerToken)) {
+      LOG.warn("Access forbidden to the add lead rest service, wrong token: {}", headerToken);
+      return Response.status(Response.Status.FORBIDDEN).entity("Access forbidden to the add lead rest service, wrong token").build();
+    }*/
```
Security Token check commented until the website update

mkrout committed:
- Code review

ahamdi reviewed (approved):
LGTM

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java:
```diff
@@ -0,0 +1,281 @@
+               lead.getLead().getId(),
+              lead.getResponse()!=null?lead.getResponse().getFormName():"");
+
+      return Response.ok("lead synchronized")
```
wouldn't a 204 be enough?

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java:
```diff
@@ -0,0 +1,281 @@
+      return Response.status(Response.Status.BAD_REQUEST).build();
+    }
+    if (lead.getLead() == null || StringUtils.isEmpty(lead.getLead().getMail())) {
+      return Response.status(Response.Status.BAD_REQUEST).entity("Lead mail needed").build();
```
A log should be added

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java:
```diff
@@ -0,0 +1,281 @@
+      return Response.status(Response.Status.FORBIDDEN).entity("Access forbidden to the add lead rest service, wrong token").build();
+    }*/
+    if (lead == null) {
+      return Response.status(Response.Status.BAD_REQUEST).build();
```
a log should be added

mkrout replied:
Ok

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java:
```diff
@@ -0,0 +1,281 @@
+
+    if (!leadCaptureSettingsService.getSettings().isCaptureEnabled()) {
+      LOG.warn("Lead capture not enabled, New lead {} not captured ", lead.getLead().getId());
+      return Response.status(Response.Status.SERVICE_UNAVAILABLE).build();
```
a 404 should be returned or any other "functionnal" status. a 503 response can be interpreted by a reverse proxy as a backend in difficulty and results to a removal from the pool of functional servers

mkrout replied:
Ok

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadCaptureSettingsRest.java:
```diff
@@ -0,0 +1,124 @@
+    try {
+      leadCaptureSettingsService.saveSettings(settings);
+      LOG.info("Lead capture settings updated by {}", sourceIdentity.getRemoteId());
+      return Response.ok("settingsUpdated", mediaType).build();
```
wouldn't a 204 be enough?

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java:
```diff
@@ -0,0 +1,281 @@
+      }
+      leadsManagementService.deleteLead(lead);
+      LOG.info("Lead {} deleted by {}", lead.getId(), sourceIdentity.getRemoteId());
+      return Response.ok().build();
```
should be a 204

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java:
```diff
@@ -0,0 +1,281 @@
+    try {
+      leadsManagementService.updateLead(lead);
+      LOG.info("Lead {} edited by {}", lead.getId(), sourceIdentity.getRemoteId());
+      return Response.ok("lead updated").build();
```
wouldn't a 204 be enough?

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java:
```diff
@@ -0,0 +1,281 @@
+    try {
+      leadsManagementService.assigneLead(lead.getId(), lead.getAssignee());
+      LOG.info("Lead {} assigned to {} by {}", lead.getId(), lead.getAssignee(), sourceIdentity.getRemoteId());
+      return Response.ok("lead assigned").build();
```
wouldn't a 204 be enough?

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/LeadsManagementRest.java:
```diff
@@ -0,0 +1,281 @@
+    try {
+      leadsManagementService.updateStatus(lead.getId(), lead.getStatus());
+      LOG.info("Lead {} status updated to {} by {}", lead.getId(), lead.getStatus(), sourceIdentity.getRemoteId());
+      return Response.ok("lead status updated").build();
```
wouldn't a 204 be enough?

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/MailTemplatesManagementRest.java:
```diff
@@ -0,0 +1,101 @@
+  public Response add(@Context UriInfo uriInfo, MailTemplateDTO templateDTO) throws Exception {
+    try {
+      mailTemplatesManagementService.addTemplate(templateDTO);
+      return Response.ok("lead synchronized").build();
```
wouldn't a 204 be enough?

vsellier commented on services/src/main/java/org/exoplatform/leadcapture/rest/MailTemplatesManagementRest.java:
```diff
@@ -0,0 +1,101 @@
+      }
+      mailTemplatesManagementService.deleteTemplate(templateEntity);
+      LOG.info("Template {} deleted by {}", id, sourceIdentity.getRemoteId());
+      return Response.ok().build();
```
Should be a 204

mkrout committed:
- Code review

mkrout committed:
- Code review

mkrout committed:
- Fix create lead without responses

mkrout committed:
- Code review

vsellier reviewed (approved):
LGTM
